R/create_protocol.R

```diff
@@ -0,0 +1,228 @@
+  parent_rmd <- file.path(path_to_protocol, paste0(protocol_filename, ".Rmd"))
+  draft(file = parent_rmd,
+        template = "template_sfp",
+        package = "protocolshelper",
+        edit = FALSE)
+  # change values in parent rmarkdown
+  original_file <- file(parent_rmd, "r")
+  original_file_content <- readLines(parent_rmd)
+  data <- list(title = title,
+               subtitle = subtitle,
+               authors = authors,
+               date = date,
+               reviewers = reviewers,
+               file_manager = file_manager,
+               revision = revision,
+               procedure = protocol_code,
+               theme = theme,
+               projectname = projectname
+  )
+  writeLines(map_chr(original_file_content,
+                     whisker.render,
+                     data),
+             parent_rmd)
+  close(original_file)
+
+  # render html
+  render(input = parent_rmd)
+}
+
+
+
+
+#' @title Function to list all occupied protocol numbers
+#'
+#' @description This function will search for protocol numbers in filenames of Rmarkdown files listed underneath the src folder. The search will be restricted to files of a given protocol type.
+#'
+#' @param protocol_type A character string equal to sfp (default), sip, sap or sop.
+#'
+#' @return A character vector with occupied protocol numbers for a specific protoocol type
+#'
+#' @importFrom rprojroot find_root is_git_root
+#' @importFrom stringr str_subset str_replace str_extract
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' get_protocolnumbers()
+#'}
+get_protocolnumbers <- function(protocol_type = c("sfp", "sip", "sap", "sop")) {
+
+  protocol_type <- match.arg(protocol_type)
+
+  project_root <- find_root(is_git_root)
+  path_to_src <- file.path(project_root, "src")
+  lf <- list.files(path = path_to_src,
+                   recursive = TRUE,
+                   full.names = FALSE,
+                   ignore.case = TRUE
+  )
+  lf <- str_subset(string = lf,
+                   pattern = "^(.*)\\/(.*)(\\.Rmd)$")
+  lf <- str_replace(string = lf,
+                    pattern = "^(.*)\\/(.*)(\\.Rmd)$",
+                    replacement = "\\2")
+  lf <- str_subset(string = lf,
+                   pattern = protocol_type)
+  lf <- str_extract(string = lf,
+                    pattern = "(?<=p-)\\d{3}")
+  lf <- lf[!is.na(lf)]
+
+  return(lf)
+}
+
+
+#' @title Function to list all short titles that are already in use.
+#'
+#' @description This function will search for short titles in filenames of Rmarkdown files listed underneath the src folder. The search will be restricted to files of a given protocol type.
+
+#' @param protocol_type A character string equal to sfp (default), sip, sap or sop.
+#'
+#' @return A character vector with short titles that are in use for a given protocol type.
+#'
+#' @importFrom rprojroot find_root is_git_root
+#' @importFrom assertthat assert_that is.string
+#' @importFrom stringr str_subset str_extract str_replace
+#'
+#' @export
```

Comment:
At least `get_short_titles()` should be exported because I refer to it in a message. The other one might also be useful at some point for the maintainers of the protocols repo.